lib/bundler/injector.rb

```diff
@@ -90,5 +115,57 @@ def append_to(gemfile_path, new_gem_lines)
         f.puts new_gem_lines
       end
     end
+
+    def remove_gems_from_gemfile(removed_deps, gemfile_path)
+      # store patterns of all gems to be removed
+      patterns = []
```

Comment:
`pattern = /gem\s+['"]#{Regexp.union(removed_deps)}\1|gem\(['"]#{Regexp.union(removed_deps)}\2\)/` ?
This will match everything in a single regular expression

Reply:
This throws an error 
```bash
TypeError: no implicit conversion of Bundler::Dependency into String
```
Looking into it.